lib/utils/report.cjs

```diff
@@ -126,6 +108,42 @@ function report(problem) {
 	result.warn(warningMessage, warningProperties);
 }
 
+/**
+ * @param {import('stylelint').Problem} problem
+ * @return {problem is import('stylelint').ProblemWithNormalizedPositions}
+ */
+function normalizeReportSourceLocation(problem) {
+	// Use the given node to calculate the start and end positions.
+	if (problem.node && (!problem.start || !problem.end)) {
+		const range = nodeRangeBy(problem.node, { index: problem.index, endIndex: problem.endIndex });
+
+		if (range) {
+			problem.start ??= range.start;
+			problem.end ??= range.end;
+		}
+	}
+
+	if (!validateTypes.isNumber(problem.line) && validateTypes.isNumber(problem.start?.line)) {
+		// Fallback the line to the line of the start position
+		problem.line = problem.start.line;
+	} else if (!problem.start && validateTypes.isNumber(problem.line)) {
```

Comment:
```suggestion
	} else if (!problem.start && problem.line) {
```

Reply:
I don't think this is meaningfully better but I also don't have a strong opinion about it. So I've applied these nits